plugins/source/hubspot/resources/plugin/client.go

```diff
@@ -63,9 +62,8 @@ func (c *Client) Sync(ctx context.Context, options plugin.SyncOptions, res chan<
 	if options.BackendOptions == nil {
 		c.logger.Info().Msg("No backend options provided, using no state backend")
 		stateClient = &state.NoOpClient{}
-		c.backendConn = nil
 	} else {
-		c.backendConn, err = grpc.DialContext(ctx, options.BackendOptions.Connection,
+		backendConn, err := grpc.DialContext(ctx, options.BackendOptions.Connection,
```

Comment:
defer close? (although it's not done anywhere else, so we might have leak here in other plugins)